python/pyspark/tests/test_taskcontext.py

```diff
@@ -181,6 +227,40 @@ def context_barrier(x):
         for pid in pids:
             self.assertTrue(pid in worker_pids)
 
+    def test_task_context_correct_with_python_worker_reuse(self):
+        """Verify the task context correct when reused python worker"""
+        # start a normal job first to start all workers and get all worker pids
+        worker_pids = self.sc.parallelize(xrange(2), 2).map(lambda x: os.getpid()).collect()
+        # the worker will reuse in this barrier job
+        rdd = self.sc.parallelize(xrange(10), 2)
+
+        def context(iterator):
+            tp = TaskContext.get().partitionId()
+            try:
+                bp = BarrierTaskContext.get().partitionId()
+            except Exception:
+                bp = -1
+
+            yield (tp, bp, os.getpid())
+
+        def verify(result, task_context_target, barrier_context_target):
```

Comment:
I think we don't need this method.

Reply:
we need `verify` three times. Too noise if remove it?

Reply:
Once we resolve https://github.com/apache/spark/pull/26239#discussion_r340931081, I think it won't be too long.

Reply:
OK. will update it.